dispatcher.py

```diff
@@ -101,37 +96,53 @@ def dispach(self):
         socket.bind(f"tcp://{self.address}:{self.port}")
         global change
         #params here are thread-safe???
-        pRSubscriber = Process(target=resultSubscriber, args=(self.uuid, len(self.urls), self.peers, self.htmls, f"{self.address}:{self.port + 1}"))
+        msg_queue = Queue()
+        args = (self.uuid, self.size, self.peers, f"{self.address}:{self.port + 1}", msg_queue)
+        pRSubscriber = Process(target=resultSubscriber, args=args)
         pRSubscriber.start()
 
         loginT = Thread(target=loginToNetwork, name="loginT", args=(self.address, self.port, self.uuid))
         loginT.start()
         #Release again when node disconnect from the network unwittingly for some reason
         lockLogin.acquire()
 
+        idx = {url:i for i, url in enumerate(self.urls)}
         while True:
             if len(self.pool) == 0:
                 #Check htmls vs urls and update pool
                 log.debug(f"Waiting for update pool of Dispatcher:{self.uuid}")
-                with lockResults:
-                #//HACK: For now the condition for the pool to be updated is that we get a result, but this is no correct because a worker can die without finish his task.
-                    #//FIXME: change is no shared between pRSubscriber process
-                    if change:
-                        log.debug(f"Updating pool of Dispatcher:{self.uuid}")
-                        #//FIXME: self.htmls is no shared between pRSubscriber process
-                        responsedURLs = {url for url, _ in self.htmls}
-                        self.pool = self.urls - responsedURLs
-                        change = False
-            try:
+                while not msg_queue.empty():
```

Comment:
From multiprocessing docs refering to Queue's methods:
empty()
Return True if the queue is empty, False otherwise. Because of multithreading/multiprocessing semantics, this is not reliable.

Reply:
Thats true. I will change it.

Solution:
Using `.get(block=False)` if the Queue is not empty return the value in the top, else raise an exception.